std/assembly/map.ts

```diff
@@ -66,8 +73,44 @@ export class Map<K,V> {
   private entriesOffset: i32 = 0;
   private entriesCount: i32 = 0;
 
-  constructor() {
-    /* nop */
+  constructor(initialEntries: MapInitialEntries<K,V> = []) {
```

Comment:
This should be nullable, and the default argument should be `null` instead of `[]`. Tip: if you wrap the entire constructor body with `if (initialEntries) {`, you can avoid doing `initialEntries!`.

(I wonder if this change could be beneficial for code that never uses this feature, by making it easier for Binaryen to optimize)